### PR TITLE
SOF-1672: Throw exception when trying to set active part as next.

### PR DESCRIPTION
**Pull request by KvelaGorrrrnio**
When trying to set an active part as next, an ActivePartException is thrown.

### CODE REVIEW

**Review by andr9528 — APPROVED** [review]

**Review by RasmusAlbrektsen — APPROVED** [review]

File: src/model/exceptions/active-part-exception.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { ErrorCode } from '../enums/error-code'
+import { Exception } from './exception'
+
+export class ActivePartException extends Exception {
```
Comment (LindvedKrvang) [inline src/model/exceptions/active-part-exception.ts:4]: Suggestion: We could call it something like `AlreadyOnAirException`. Then the exception is a bit more reuseable and we don't risk as much having exceptions only use for a single specific things

File: src/model/entities/test/rundown.spec.ts [hunk 2]
```diff
@@ -2805,12 +2804,48 @@ describe(Rundown.name, () => {
...
+          isRundownActive: true,
+          alreadyActiveProperties: {
+            activeCursor: {
+              segment: activeSegment,
+              part: activePart,
+            },
+            nextCursor: {
+              segment: nextSegment,
+              part: nextPart,
+            },
```
Comment (LindvedKrvang) [inline src/model/entities/test/rundown.spec.ts:2836]: You don't need a `nextCursor` for this test so you can skip all the setup for the nextCursor.

File: src/model/entities/test/rundown.spec.ts [hunk 3]
```diff
@@ -2805,12 +2804,48 @@ describe(Rundown.name, () => {
...
       testee.setNext(activeSegment.id, otherPartInActiveSegment.id)
 
       verify(mockedNextPart.reset()).once()
     })
+
+    describe('when next part is on air', () => {
+      it('throws an active part exception', () => {
+        const activePart: Part = EntityMockFactory.createPart({ id: 'active-part-id', isOnAir: true })
+        const mockedActiveSegment: Segment = EntityMockFactory.createSegmentMock({ id: 'active-segment-id', parts: [activePart] })
+        when(mockedActiveSegment.findPart(activePart.id)).thenReturn(activePart)
```
Comment (LindvedKrvang) [inline src/model/entities/test/rundown.spec.ts:2822]: Instead of mocking you could also use the `EntityTestFactory`.

**Review by LindvedKrvang — APPROVED** [review]

File: src/model/exceptions/active-part-exception.ts (hunk 1 above)
Comment (KvelaGorrrrnio) [reply src/model/exceptions/active-part-exception.ts:4]: Do you want `ActivePartException` and `ActiveRundownException` merged together to `AlreadyOnAirException`?

File: src/model/entities/test/rundown.spec.ts (hunk 2 above)
Comment (KvelaGorrrrnio) [reply src/model/entities/test/rundown.spec.ts:2834]: There is a check in rundown.setNext that throws if no nextCursor is set.

File: src/model/entities/test/rundown.spec.ts (hunk 3 above)
Comment (KvelaGorrrrnio) [reply src/model/entities/test/rundown.spec.ts]: Updated to using the EntityTestFactory.

File: src/model/exceptions/active-part-exception.ts (hunk 1 above)
Comment (LindvedKrvang) [reply src/model/exceptions/active-part-exception.ts:4]: Initial thoughs: no. A Rundown being active doesn't necessarily means that it is on Air. You can have an active Rundown that has no Parts on Air (when we activate the Rundown before doing a Take)

Comment (KvelaGorrrrnio) [reply src/model/exceptions/active-part-exception.ts]: I have changed it to `OnAirException`. (ActiveRundown is still kept)

**Pull request merged**